addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -300,8 +300,8 @@ var FieldMany2One = AbstractField.extend({
     _renderReadonly: function () {
         var value = _.escape((this.m2o_value || "").trim()).split("\n").join("<br/>");
         this.$el.html(value);
-        if (!this.nodeOptions.no_open) {
-            this.$el.attr('href', '#');
+        if (!this.nodeOptions.no_open && this.value) {
+            this.$el.attr('href', _.str.sprintf('#id=%s&model=%s', this.value.res_id, this.field.relation));
```

Comment:
could we add the current menu_id as well? Even if it is not the real menu to lead to that record, it allows to have the same top menu as on the previous page

Reply:
After discussion with @ged-odoo we keep it simple as getting current menu_id not simple for instance